File: curso python/ejemplos python/problema20.py
"""determinar el monto de descuento,el importe del impuesto y el importe a pagar

consumo            descuento %
hasta 100           10
mayor a 100         20
mayor a 200         30
"""

D1= 0.1
D2 = 0.2
D3= 0.3
IGV= 0.19  # 19%
consumo = float (input ('introduce la cantidad de consumo: '))
if consumo <= 100:
    descuento = consumo *D1
elif consumo> 100 and consumo <200:
    descuento = consumo * D2
elif consumo >=200:
    descuento = consumo * D3

impuesto = (consumo - descuento) * IGV
importe =  (consumo - descuento)+ impuesto
print (f'monto del descuento: {descuento}')
print (f'impuesto con IGV: {impuesto}')
print (f"importe total a pagar:{importe} ")

File: curso c++/ejemplos/problema20.cpp
//determinar el monto de descuento,el importe del impuesto y el importe a pagar
/*
consumo            descuento %
hasta 100           10
mayor a 100         20
mayor a 200         30

*/
#include <iostream>
using namespace std;
int main (){
    const float D1=0.1;
    const float D2 = 0.2;
    const float D3 = 0.3;
    float igv= 0.19;
    float des,impuesto,importe,consumo;
    cout << "inserte ele monto: ";
    cin >> consumo;

    if (consumo > 200)
        des=consumo * D3;
    else
        if (consumo >100)
            des= consumo * D2;
        else 
            des = consumo * D1;
    
    impuesto = (consumo - des) * igv;
    importe = (consumo - des)+impuesto;

    cout << "monto descuento: "<<des<<endl;
    cout << "impuesto igv: "<<impuesto<<endl;
    cout << "importe total :"<<importe;

    return 0;
}